csrc/device_lower/pass/allocation.cpp

```diff
@@ -163,13 +163,17 @@ class AllocationInserter : public kir::ExprMutator {
     std::vector<Val*> alloc_dims;
 
     for (const auto id : maybe_rfactor_domain) {
-      if (id->isReduction() || id->isStride() || id->isDeviceDim()) {
+      if (id->isReduction() || id->isStride()) {
         continue;
       } else if (id->isBroadcast()) {
         // No matter whether this broadcast is expanded or not, we always
         // allocate size 1
         alloc_dims.emplace_back(id->container()->oneVal());
         continue;
+      } else if (id->isDeviceDim()) {
```

Comment:
Also, I don't remember what was our conclusion about allocating with size 1 vs. not allocating..?
I see that allocating with size 1 implies to squeeze/unsqueeze the tensors, however I can't remember what is the benefit

Reply:
We might need the device dimension allocated to handle splits.
```
tv0 = makeContig(2); // (n, m)
tv0->axis(0)->split(ndevices, false); // (ndevices, n//ndevices, m)
tv0->axis(0)->parallelize(DIDx);
```
will allocate a 2D tensor with size (n//ndevices, m)

Reply:
Why would it help then to have (1, n//ndevices, m) instead of (n//ndevices, m) ? I am not sure to understand why it is necessary

Reply:
I was thinking we allocate the same number of dimensions as the tensorview. So the above example would allocate (n//ndevices, m). 

```
tv0 = makeContig(2); // (n, m)
tv0->axis(0)->parallelize(DIDx);
```
Then, this case, which assumes n == ndevices, would allocate (1, m) instead of (m).